x-pack/plugins/apm/server/lib/search_strategies/queries/field_stats/get_numeric_field_stats.ts

```diff
@@ -87,7 +87,7 @@ export const fetchNumericFieldStats = async (
   );
   const { body } = await esClient.search(request);
 
-  const aggregations = body.aggregations as {
+  const aggregations = body.aggregations as unknown as {
```

Comment:
`tsc` complains `sample` doesn't exist on `Record<string, AggregationsAggregate>`. We need to refactor the way `elasticsearch-js` declares `aggregations type. A first proposal is here https://github.com/elastic/elasticsearch-js/pull/1596

Reply:
Very likely this new aggregations work will land only in v8.

Reply:
it can create some merge conflicts during a bug fix backporting, but let's see